exec/cnex/cell.c

```diff
@@ -239,6 +247,71 @@ BOOL cell_arrayElementExists(const Cell *a, const Cell *e)
     return FALSE;
 }
 
+TString *cell_toString(Cell *c)
+{
+    assert(c->type != cNothing);
+    TString *r = string_newString();
+
+    switch (c->type) {
+        case cAddress:
+            break;
+        case cArray:
+        {
+            size_t x, y;
+            /* ToDo: Fix this to use proper string appends!! */
+            char *s = malloc(STRING_BUFFER_SIZE);
+            if (s == NULL) {
+                fatal_error("Could not allocate %d byte string buffer for array", STRING_BUFFER_SIZE);
+            }
+            strcpy(s, "[");
+            for (x = 0; x < c->array->size; x++) {
+                if (strlen(s) > 1) {
+                    strcat(s, ", ");
+                }
+                if (c->array->data[x].type == cString) {
```

Comment:
I'm not sure why this needs to look into the type of each element. This should simply recurse and append the result of cell_toString() for each cell. Right now, you wouldn't be able to handle an array of arrays, for example.

Reply:
Fixed by only checking for `cString` so quotes can be placed around the string.  All other elements are simply the result of a recursion into `cell_toString()`